pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -234,6 +419,7 @@ func (p *Propagator) processEvent() {
 			return
 		}
 		p.removeNode(existingNode)
+		// this read of "dependents" doesn't require a RLock, because it's in processEvent(), which is single-threaded and is the only writer of "dependents".
```

Comment:
I really don't think this follows-- someone else (a test) could have already gotten a pointer to existingNode.